pkg/volume/util/operationexecutor/operation_executor.go

```diff
@@ -793,15 +826,95 @@ func (oe *operationExecutor) generateUnmountDeviceFunc(
 		markDeviceUnmountedErr := actualStateOfWorld.MarkDeviceAsUnmounted(
 			deviceToDetach.VolumeName)
 		if markDeviceUnmountedErr != nil {
-			// On failure, just log and exit. The controller will retry
-			glog.Errorf(
+			// On failure, return error. Caller will log and retry.
+			return fmt.Errorf(
 				"MarkDeviceAsUnmounted failed for device %q (spec.Name: %q) with: %v",
 				deviceToDetach.VolumeName,
 				deviceToDetach.VolumeSpec.Name(),
 				markDeviceUnmountedErr)
-			return markDeviceUnmountedErr
 		}
 
 		return nil
 	}, nil
 }
+
+func (oe *operationExecutor) generateVerifyControllerAttachedVolumeFunc(
+	volumeToMount VolumeToMount,
+	nodeName string,
+	actualStateOfWorld ActualStateOfWorldAttacherUpdater) (func() error, error) {
+	return func() error {
+		if !volumeToMount.PluginIsAttachable {
+			// If the volume does not implement the attacher interface, it is
+			// assumed to be attached and the the actual state of the world is
+			// updated accordingly.
+			addVolumeNodeErr := actualStateOfWorld.MarkVolumeAsAttached(
+				volumeToMount.VolumeSpec, nodeName, volumeToMount.DevicePath)
+			if addVolumeNodeErr != nil {
+				// On failure, return error. Caller will log and retry.
+				return fmt.Errorf(
+					"VerifyControllerAttachedVolume.MarkVolumeAsAttached failed for volume %q (spec.Name: %q) pod %q (UID: %q) with: %v.",
```

Comment:
Just want to make sure, this error will be returned as func() error, or the second error?
'